iroh-net/src/magicsock.rs

```diff
@@ -1121,29 +1204,30 @@ impl MagicSock {
             closing: AtomicBool::new(false),
             closed: AtomicBool::new(false),
             derp_recv_receiver,
-            network_recv_wakers: std::sync::Mutex::new(None),
-            network_send_wakers: std::sync::Mutex::new(None),
+            network_recv_wakers: parking_lot::Mutex::new(None),
```

Comment:
why parking lot? Perf wise the stdlib ones are comparable since some time. Because of the deadlock detection feature?

Reply:
Actually no real reason, just that it cannot panic? Can change back to `std`. We have a mix of both mutexes in here, maybe we should settle for one.

Reply:
I used parking_lot in a couple of places because it has better support for mapping locks. That's the only reason I found to use them over std

Reply:
It used to be that parking lot was insanely fast compared to stdlib. And of course that you don't have to do the annoying unwrap everywhere. But the stdlib perf has catched up, so these days most of the time I use std. But the deadlock detector of parking lot is neat.